### PR TITLE
compute_ctl: start pg in cgroup for vms

### PR DESCRIPTION
**Pull request by fprasx**
## Summary of changes
Starts `postgres` in cgroup directly from `compute_ctl` instead of from `vm-builder`. This is required because the `vm-monitor` cannot be in the cgroup it is managing. Otherwise, it itself would be frozen when freezing the cgroup.

Requires https://github.com/neondatabase/cloud/pull/6331, which adds the `AUTOSCALING` environment variable letting `compute_ctl` know to start `postgres` in the cgroup.

Requires https://github.com/neondatabase/autoscaling/pull/468, which prevents `vm-builder` from starting the monitor and putting postgres in a cgroup. This will require a `VM_BUILDER_VERSION` bump.

**EDIT**: dropping this change for the time being as the cplane overwrites the entrypoint vm-builder sets for the images it produces. Thus, `compute_ctl` isn't actually in the cgroup. Once the monitor merged, we can add postgres back into the cgroup.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: compute_tools/src/compute.rs [hunk 1]
```diff
@@ -591,10 +605,17 @@ impl ComputeNode {
 
         // Start postgres
         info!("starting postgres");
-        let mut pg = Command::new(&self.pgbin)
-            .args(["-D", pgdata])
-            .spawn()
-            .expect("cannot start postgres process");
+
+        // VMs have cg-exec in their environment while pods don't
+        let vm = Command::new("/usr/bin/cgexec").output().is_ok();
```
[the remark this answers is not in the document]
Comment (fprasx) [reply compute_tools/src/compute.rs]: https://github.com/neondatabase/cloud/pull/6331. This sets the `AUTOSCALING` environment variable for VM computes. Do you think is an ok way to do it?

[the remark this answers is not in the document]
Comment (vadim2404) [reply compute_tools/src/compute.rs]: initially, I thought about changing https://github.com/neondatabase/neon/blob/main/Dockerfile.compute-node (and it's relevant version for VMs). But, I'm not against your proposal

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -804,7 +804,9 @@ jobs:
 
       - name: Downloading vm-builder
         run: |
-          curl -fL https://github.com/neondatabase/autoscaling/releases/download/$VM_BUILDER_VERSION/vm-builder -o vm-builder
+          # Custom vm-builder since there has not yet been a release
+          # felix-monitor-testing on aws
+          curl -fL ec2-18-217-40-54.us-east-2.compute.amazonaws.com:8000/vm-builder -o vm-builder
```
Comment (vadim2404) [inline .github/workflows/build_and_test.yml:809]: Is the comment relevant?

File: compute_tools/src/compute.rs [hunk 3]
```diff
@@ -180,6 +181,24 @@ impl TryFrom<ComputeSpec> for ParsedSpec {
     }
 }
 
+/// Spawns a command in the neon-postgres cgroup if it exists, otherwise just
+/// spawns it normally, returning a handle to it.
```
Comment (sharnoff) [inline compute_tools/src/compute.rs:185]: comment is outdated? (spawns in neon-postgres if _autoscaling_, ignoring whether the cgroup exists)

File: compute_tools/src/compute.rs [hunk 4]
```diff
@@ -180,6 +181,24 @@ impl TryFrom<ComputeSpec> for ParsedSpec {
...
+///
+/// This function should be used to start postgres, as it will start in in the
+/// neon-postgres cgroup if we are a VM. The cgroup will exist in VM's because
+/// vm-builder creates it during the sysinit phase of its inittab.
+fn maybe_cgexec(cmd: &str) -> Command {
+    // The cplane sets this env var for autoscaling computes
+    if env::var("AUTOSCALING").is_ok() {
```
Comment (sharnoff) [inline compute_tools/src/compute.rs:192]: consider `env::var_os(...).is_some()` ?

**Review by sharnoff — COMMENTED** [review]
Broadly looks ok. The changes to `build_and_test.yml` are temporary, for testing, right?

Searched for instances of `Command::new` in `compute_tools/src` and noticed this:

https://github.com/neondatabase/neon/blob/cc2d00fea4952d30865a193648fdad4e0574f1ef/compute_tools/src/compute.rs#L457-L460

Does it need to be included in `neon-postgres` as well?

**Comment by fprasx** [comment]
Yup, once https://github.com/neondatabase/autoscaling/pull/468 is merged, we'll just used that version of `vm-builder`

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (fprasx) [reply .github/workflows/build_and_test.yml:809]: Just explaining the temporary change. See comments below for more context.

File: compute_tools/src/compute.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (fprasx) [reply compute_tools/src/compute.rs]: I added the environment variable in neondatabase/cloud#6354

**Review by sharnoff — APPROVED** [review]

**Pull request merged**